<a href="https://colab.research.google.com/github/jdmartinezrs/incomesAndBillsProject/blob/main/Mision1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import files

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
#Importación dataset control maquinaria
import pandas as pd
df = pd.read_excel("/content/drive/MyDrive/datasets mision1/controlmaquinaria.xlsx")
df.shape

(70, 41)

In [4]:
#DatasetControl maquinaria
df

Unnamed: 0                       Unnamed: 1       Unnamed: 2 Unnamed: 3  \
0          NaN                              NaN              NaN        NaN   
1          NaN  CONTROL DE FLOTA DE TRANSPORTES              NaN        NaN   
2          NaN                              NaN              NaN        NaN   
3          NaN                              NaN              NaN        NaN   
4          NaN                             Item           Codigo      Placa   
..         ...                              ...              ...        ...   
65         NaN                               61  VE-TB-HY-WGA173     WGA173   
66         NaN                               62  VE-TB-MD-TBK246     TBK246   
67         NaN                               63  VE-VE-RE-HHP403     HHP403   
68         NaN                               64  VE-VE-RE-MUZ920     MUZ920   
69         NaN                               65  VE-VS-CH-TAV332     TAV332   

             Unnamed: 4 Unnamed: 5 Unnamed: 6        Unnamed: 7  \
0                   NaN        NaN        NaN               NaN   
1          FECHA ACTUAL        NaN        NaN               NaN   
2                   NaN        NaN        NaN               NaN   
3   2025-05-21 00:00:00        NaN        NaN               NaN   
4         Tipo Vehiculo      Marca        Año             Linea   
..                  ...        ...        ...               ...   
65                TURBO    HYUNDAI       2014              HD65   
66                TURBO      MAZDA       2000              T 45   
67             VEHICULO    RENAULT       2016        CLIO STYLE   
68             VEHICULO    RENAULT       2014  LOGAN EXPRESSION   
69    VOLQUETA SENCILLA  CHEVROLET       2012               FVR   

                                           Unnamed: 8  \
0                                                 NaN   
1   CONTROL DE MAQUINARIA, VEHICULOS Y CONDUCTORES...   
2                                                 NaN   
3                                        Revision: 04   
4                                         Propietario   
..                                                ...   
65                             PAVIMENTOS ANDINOS S.A   
66                             PAVIMENTOS ANDINOS S.A   
67                             PAVIMENTOS ANDINOS S.A   
68                             PAVIMENTOS ANDINOS S.A   
69                                               MGAG   

                             Unnamed: 9  ... Unnamed: 31 Unnamed: 32  \
0                                   NaN  ...         NaN         NaN   
1                                   NaN  ...         NaN         NaN   
2                                   NaN  ...         NaN         NaN   
3   Fecha de Revision: 30 abril de 2025  ...         NaN         NaN   
4                             Conductor  ...    AlertaPG    EstadoPG   
..                                  ...  ...         ...         ...   
65                             CASTILLA  ...         NaN         NaN   
66                        SIN CONDUCTOR  ...         NaN         NaN   
67                             CASTILLA  ...         NaN         NaN   
68                             CASTILLA  ...         NaN         NaN   
69                      MARLON MARTINEZ  ...         NaN         NaN   

          Unnamed: 33      Unnamed: 34      Unnamed: 35     Unnamed: 36  \
0                 NaN              NaN              NaN             NaN   
1                 NaN              NaN              NaN             NaN   
2                 NaN              NaN              NaN             NaN   
3                 NaN              NaN              NaN             NaN   
4   Vigencia Extintor  Alerta Extintor  Estado Extintor  Otros Permisos   
..                ...              ...              ...             ...   
65                NaN              NaN              NaN             NaN   
66                NaN              NaN              NaN             NaN   
67                NaN    

In [45]:
#Dataset de mantenimiento
import pandas as pd
dt = pd.read_excel("/content/drive/MyDrive/datasets mision1/mantenimiento.xlsm")
dt.shape

(57, 17)

In [46]:
#Dataset de mantenimiento
dt

Unnamed: 0 Unnamed: 1  \
0                          NaN        NaN   
1          2025-05-21 00:00:00        NaN   
2                          NaN      #REF!   
3                          NaN        NaN   
4                        Fecha       ITEM   
5                          NaN          1   
6                          NaN          2   
7                          NaN          3   
8   2025-05-21 11:57:03.744000          4   
9   2025-05-21 11:57:03.744000          5   
10  2025-05-21 11:57:03.744000          6   
11  2025-05-21 11:57:03.744000          7   
12  2025-05-21 11:57:03.744000          8   
13  2025-05-21 11:57:03.744000          9   
14  2025-05-21 11:57:03.744000         10   
15  2025-05-21 11:57:03.744000         11   
16  2025-05-21 11:57:03.744000         12   
17  2025-05-21 11:57:03.744000         13   
18  2025-05-21 11:57:03.744000         14   
19  2025-05-21 11:57:03.744000         15   
20  2025-05-21 11:57:03.744000         16   
21  2025-05-21 11:57:03.744000         17   
22  2025-05-21 11:57:03.744000         18   
23  2025-05-21 11:57:03.744000         19   
24  2025-05-21 11:57:03.744000         20   
25  2025-05-21 11:57:03.744000         21   
26                         NaN         22   
27                         NaN         23   
28                         NaN         24   
29  2025-05-21 11:57:03.744000         25   
30  2025-05-21 11:57:03.744000         26   
31  2025-05-21 11:57:03.744000         27   
32  2025-05-21 11:57:03.744000         28   
33                         NaN         29   
34                         NaN         30   
35  2025-05-21 11:57:03.744000         31   
36  2025-05-21 11:57:03.744000         32   
37  2025-05-21 11:57:03.744000         32   
38  2025-05-21 11:57:03.744000        NaN   
39  2025-05-21 11:57:03.744000        NaN   
40  2025-05-21 11:57:03.744000        NaN   
41  2025-05-21 11:57:03.744000        NaN   
42                         NaN        NaN   
43                         NaN        NaN   
44                         NaN        NaN   
45                         NaN        NaN   
46                         NaN        NaN   
47                         NaN        NaN   
48                         NaN        NaN   
49                         NaN        NaN   
50                         NaN        NaN   
51                         NaN        NaN   
52                         NaN        NaN   
53                         NaN        NaN   
54                         NaN        NaN   
55                         NaN        NaN   
56                         NaN        NaN   

                                           Unnamed: 2        Unnamed: 3  \
0                                                 NaN               NaN   
1                                                 NaN               NaN   
2                                               #REF!               NaN   
3   MANTENIMIENTO DE LUBRICACION MOTOR y REVISION ...               NaN   
4                                    CODIGO DE EQUIPO  Tipo de Vehiculo   
5                                     VE-DT-CH-XMD284      CARRO TANQUE   
6                                     VE-DT-FL-XMD588       DOBLETROQUE   
7                                     VE-DT-FL-SUE755       DOBLETROQUE   
8                                     VE-DT-FL-WFD547       DOBLETROQUE   
9                                     VE-DT-IN-TAV909       DOBLETROQUE   
10                                    VE-DT-IN-TAW674       DOBLETROQUE   
11                                    VE-DT-IN-TAW675       DOBLETROQUE   
12                                    VE-DT-IN-TAW676       DOBLETROQUE   
13                                    VE-DT-IN-TAW677       DOBLETROQUE   
14                                    VE-DT-IN-TAW678       DOBLETROQUE   
15                                    VE-DT-IN-TTR442       DOBLETROQUE   
16                                    VE-DT-IN-WFC325       DOBLETROQUE   
17                                    VE-DT-IN-WFC326      

In [87]:
from sklearn.linear_model import LinearRegression
import numpy as np

In [64]:
dt = pd.read_excel("/content/drive/MyDrive/datasets mision1/mantenimiento.xlsm", header=5)

In [81]:
print(dt.head(2))

  Fecha  ITEM CODIGO DE EQUIPO Tipo de Vehiculo fecha de revision  \
0   NaT   1.0  VE-DT-CH-XMD284     CARRO TANQUE        2025-05-21   
1   NaT   2.0  VE-DT-FL-XMD588      DOBLETROQUE        2025-05-21   

   Km/ Hr Plan Fecha Ult Pre-Operacional  Dias Pend  Ult Km/Hr  Km/Hr Cambio  \
0      10000.0                2025-05-20        1.0   320832.0      316580.0   
1      10000.0                2023-09-21      608.0   209543.0      199850.0   

  Fecha Cambio Fecha Prox Cambio Unnamed: 12  Km/Hr Proyectada  Km/Hr Pend  \
0   2024-06-21        2025-01-17  2025-08-15          326580.0      5748.0   
1   2023-07-11               NaT         NaT          209850.0       307.0   

   Progreso Observaciones   
0    0.5748            NaN  
1    0.0307            NaN  


In [91]:
from sklearn.impute import SimpleImputer
import numpy as np

# Create an imputer that replaces NaN with the mean of the column
# You can choose other strategies like 'median' or 'most_frequent'
imputer = SimpleImputer(missing_values=np.nan, strategy='mean')

# Fit the imputer to the data and transform the columns
dt['Km/Hr Pend'] = imputer.fit_transform(dt[['Km/Hr Pend']])
dt['Ult Km/Hr'] = imputer.fit_transform(dt[['Ult Km/Hr']])

# Now define y and x using the imputed data
y = dt['Km/Hr Pend'].values.reshape(-1, 1)
x = dt['Ult Km/Hr'].values.reshape(-1, 1)

# Proceed with your LinearRegression model
model = LinearRegression()
model.fit(x, y)

# Print coefficients, make predictions, etc.
print("Coeficiente:", model.coef_)
print("Intercepción:", model.intercept_)

Coeficiente: [[0.00312893]]
Intercepción: [5427.17926385]


**Coeficiente: [[0.00312893]]**

Este valor (0.00312893) representa la pendiente de la línea de regresión.
Significa que, por cada aumento de una unidad en Ult Km/Hr, se espera un aumento de aproximadamente 0.00312893 unidades en Km/Hr Pend.
En el contexto de tu data, esto sugiere una relación lineal positiva, aunque pequeña. A medida que los kilómetros/horas registrados (Ult Km/Hr) aumentan, los kilómetros/horas pendientes (Km/Hr Pend) tienden a aumentar ligeramente.

**Intercepción: [5427.17926385]**

Este valor (5427.17926385) representa el punto donde la línea de regresión cruza el eje Y (cuando x es igual a 0).
En este caso, cuando Ult Km/Hr es 0, se espera que Km/Hr Pend sea aproximadamente 5427.18.
Sin embargo, es importante ser cauteloso con la interpretación de la intercepción si un valor de Ult Km/Hr de 0 no tiene sentido en el contexto de tus datos (por ejemplo, si nunca tienes un vehículo con 0 Kilómetros/Horas registrados). A veces, la intercepción es más un ajuste matemático para la línea de regresión.

In [93]:
# Define the coefficients and intercept from your model
coeficiente = 0.00312893
intercepcion = 5427.17926385

In [94]:
# Define a variable for the input value (Ult Km/Hr)
# Replace 'some_ult_km_hr_value' with the actual value you want to predict for
ult_km_hr_value = 300000  # Example value

# Calculate the predicted Km/Hr Pend using the equation
km_hr_pend_predicho = (coeficiente * ult_km_hr_value) + intercepcion

# Print the predicted value
print("Km/Hr Pend predicho:", km_hr_pend_predicho)

Km/Hr Pend predicho: 6365.85826385


El resultado Km/Hr Pend predicho: 6365.85826385 significa que, según tu modelo de regresión lineal, para un vehículo con 300,000 Kilómetros/Horas registrados (Ult Km/Hr), la cantidad esperada de Kilómetros/Horas pendientes (Km/Hr Pend) es aproximadamente 6365.86.

Contexto: La interpretación de este valor debe hacerse dentro del contexto de tus datos y lo que representan Km/Hr Pend y Ult Km/Hr. Si Km/Hr Pend indica cuánto falta para un próximo mantenimiento, entonces la predicción sugiere que un vehículo con 300,000 Km/Hr tiene aproximadamente 6365.86 Km/Hr restantes antes de su próximo mantenimiento.

In [75]:
print(dt[['Tipo de Vehiculo', 'fecha de revision']])

   Tipo de Vehiculo fecha de revision
0      CARRO TANQUE        2025-05-21
1       DOBLETROQUE        2025-05-21
2       DOBLETROQUE        2025-05-21
3       DOBLETROQUE        2025-05-21
4       DOBLETROQUE        2025-05-21
5       DOBLETROQUE        2025-05-21
6       DOBLETROQUE        2025-05-21
7       DOBLETROQUE        2025-05-21
8       DOBLETROQUE        2025-05-21
9       DOBLETROQUE        2025-05-21
10      DOBLETROQUE        2025-05-21
11      DOBLETROQUE        2025-05-21
12      DOBLETROQUE        2025-05-21
13      DOBLETROQUE        2025-05-21
14      DOBLETROQUE        2025-05-21
15      DOBLETROQUE        2025-05-21
16      DOBLETROQUE        2025-05-21
17      DOBLETROQUE        2025-05-21
18      DOBLETROQUE        2025-05-21
19             GRUA        2025-05-21
20        IRRIGADOR        2025-05-21
21     TRACTOCAMION        2025-05-21
22     TRACTOCAMION        2025-05-21
23     TRACTOCAMION        2025-05-21
24            TURBO        2025-05-21
25          

In [73]:
print(dt['CODIGO DE EQUIPO'])

0     VE-DT-CH-XMD284
1     VE-DT-FL-XMD588
2     VE-DT-FL-SUE755
3     VE-DT-FL-WFD547
4     VE-DT-IN-TAV909
5     VE-DT-IN-TAW674
6     VE-DT-IN-TAW675
7     VE-DT-IN-TAW676
8     VE-DT-IN-TAW677
9     VE-DT-IN-TAW678
10    VE-DT-IN-TTR442
11    VE-DT-IN-WFC325
12    VE-DT-IN-WFC326
13    VE-DT-IN-WFC327
14    VE-DT-KW-TAX535
15    VE-DT-KW-TAX539
16    VE-DT-KW-TAX540
17    VE-DT-KW-TTS952
18    VE-DT-FL-SOZ243
19    VE-GR-FO-WFG316
20    VE-TB-MD-TBK246
21    VE-TM-FL-XMC858
22    VE-TM-HN-SRO974
23    VE-TM-FL-SVO546
24    VE-TB-CH-SRL418
25    VE-TB-CH-SUF310
26    VE-TB-HY-WGA173
27    VE-TB-CH-SUF685
28    VE-TB-CH-TAX650
29    VE-TB-FT-WOM542
30    VE-VS-CH-TAV332
31    VE-TM-FL-WFL031
32                NaN
33                NaN
34                NaN
35                NaN
Name: CODIGO DE EQUIPO, dtype: object


In [51]:
data = pd.DataFrame(dt)
for i in range(1, 2):
  print(dt.iloc[0])


Unnamed: 0                                                                               Fecha
Unnamed: 1                                                                                ITEM
MANTENIMIENTO DE LUBRICACION MOTOR y REVISION DE FALLAS GENERALES             CODIGO DE EQUIPO
Unnamed: 3                                                                    Tipo de Vehiculo
Unnamed: 4                                                                   fecha de revision
Unnamed: 5                                                                         Km/ Hr Plan
Unnamed: 6                                                           Fecha Ult Pre-Operacional
Unnamed: 7                                                                           Dias Pend
Unnamed: 8                                                                           Ult Km/Hr
Unnamed: 9                                                                        Km/Hr Cambio
Unnamed: 10                                       

In [19]:
df = pd.read_excel("/content/drive/MyDrive/datasets mision1/controlmaquinaria.xlsx", header=4)


In [22]:
df.dropna(how='all', inplace=True)

In [40]:
datos = pd.DataFrame(df)
#for i in range(1, 3):
print(df.iloc[0])
#print(df.iloc[0:3])
#print(df.loc[0, ['Marca', 'Año']])

#print(df.iloc[0, [0,1,2,3,4,5,6,7,8,9,10,11,12,13]])


Unnamed: 0                                                   NaN
Unnamed: 1                                                  Item
Unnamed: 2                                                Codigo
Unnamed: 3                                                 Placa
2025-05-21 00:00:00                                Tipo Vehiculo
Unnamed: 5                                                 Marca
Unnamed: 6                                                   Año
Unnamed: 7                                                 Linea
Revision: 04                                         Propietario
Fecha de Revision: 30 abril de 2025                    Conductor
Unnamed: 10                                   Ubicación/Proyecto
Pagina 1 de 1                                   Proveedor Poliza
Unnamed: 12                                            Poliza TR
Unnamed: 13                                            Poliza RC
Unnamed: 14                                         Costo Poliza
Unnamed: 15              

una variable dependiente (y) que queremos predecir, y variables independientes (x) que usaremos para hacer la predicción.